src/StateContext.php

```diff
@@ -1,25 +1,30 @@
 <?php
 /**
- * This file is part of the status project.
+ * This file is part of the php-state project.
  *
- * (c) Yannick Voyer (http://github.com/yvoyer)
+ * (c) Yannick Voyer <star.yvoyer@gmail.com> (http://github.com/yvoyer)
  */
 
 namespace Star\Component\State;
 
-/**
- * Class StateContext
- *
- * @author  Yannick Voyer (http://github.com/yvoyer)
- *
- * @package Star\Component\State
- */
 interface StateContext
 {
     /**
-     * This method SHOULD only be called by the State implementation.
-     *
      * @param State $state
+     * @internal Public to be used by the State machine only.
      */
     public function setState(State $state);
+
+    /**
+     * @return State
+     * @internal Public to be used by the State machine only.
+     */
+    public function getCurrentState();
+
+    /**
+     * Returns the alias used to identify custom events for transitions.
+     *
+     * @return string
+     */
+    public function contextAlias();
```

Comment:
Could add an interface for a static method that encapsulates the creation of states and transitions in order to register subscribers in the app level, while being able to centralize it on the object.